CONTRIBUTING.md

```diff
@@ -0,0 +1,287 @@
+
+5. **Make your changes** according to our [code standards](https://eest.ethereum.org/main/getting_started/code_standards).
+
+6. **For EVM Tests**: Review the cases in the [EIP checklist template](./docs/writing_tests/checklist_templates/eip_testing_checklist_template.md).
+
+7. **Verify your changes** by running the appropriate checks:
+
+    ```bash
+    uvx --with=tox-uv tox -e lint,typecheck
+    ```
+
+8. **Commit your changes** with meaningful commit messages (see [Commit Messages, Issues and PR Titles](#commit-messages-issue-and-pr-titles)).
+
+9. **Push your branch** to your GitHub fork:
+
+    ```bash
+    git push -u origin your-branch-name
+    ```
+
+10. **Create a pull request** by navigating to your fork on GitHub and clicking the "New Pull Request" button.
+
+### Branch Naming Conventions
+
+Branch names should follow this format:
+
+```text
+<type>/<short-description>
+```
+
+Where `<type>` matches the semantic commit types:
+
+- `feat/` - For new features.
+- `fix/` - For bug fixes.
+- `docs/` - For documentation changes.
+- `test/` - For adding or modifying tests.
+- `refactor/` - For code refactoring.
+- `chore/` - For maintenance tasks.
+
+Examples:
+
+```text
+feat/add-cancun-blob-tests
+fix/prague-consume-genesis
+docs/improve-installation-guide
+```
+
+### PR Review Process
+
+1. **Initial checks**: When you submit a PR, automated CI checks will run. Make sure all checks pass before requesting a review.
+
+2. **Requesting review**: Feel free to tag a maintainer or ask for review in a PR comment.
+
+3. **Review feedback**: Maintainers will review your code and may suggest changes. Please address all comments and engage in discussion if needed.
+
+4. **Iteration**: Make requested changes, push to your branch, and the PR will update automatically. No need to create a new PR.
+
+5. **Approval**: Once your changes are approved, a maintainer will merge your PR.
+
+### PR Expectations
+
+To increase the chances of your PR being merged quickly:
+
+- **Scope**: Keep PRs focused on a _single issue or feature_.
+- **CI checks**: Ensure all CI checks pass before requesting review; but do ask for help if you don't understand the fail!
+- **Clean history**: Use meaningful, atomic commits that can be easily understood.
+- **Tests**: Include tests for new functionality.
+- **Documentation**: Update documentation for new features or changes.
+- **Responsiveness**: Try to respond to review feedback within a reasonable time.
+
+### Code Standards and Enforced CI Checks
+
+We enforce lint, code formatting and unit test checks in our CI - for detailed code standards and enforcement checks, see our [Code Standards documentation](https://eest.ethereum.org/main/getting_started/code_standards).
+
+### Commits
+
+It's recommended to keep changes logically grouped into smaller, individual commits to make changes easier to review.
+
+### Commit Messages, Issue and PR Titles
+
+We use semantic commit messages and PR titles following the format:
```

Comment:
Do we follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)? Using [`!` for breaking change](https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with--to-draw-attention-to-breaking-change) could be helpful.
```
feat(eest)!: add new test generator command
```

Reply:
We should add this and adhere to the standard, thanks! I've always just gone by https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716

Reply:
Added in [0c596cf](https://github.com/ethereum/execution-spec-tests/pull/1604/commits/0c596cfdf18f2702093318898111c052786926b2).